### PR TITLE
Support customization of link text

### PR DESCRIPTION
**Pull request by yanis-sadeg**
## Description
Added functionality to add text to the location button. 
It is activated with the "text" option and does not allow HTML characters.

Tested on Firefox, Chrome, Safari. ✔️

## Example
```
lc = L.control.locate({
    strings: {
        title: "Show me where I am, yo!",
        text: "Locate me"
    }
}).addTo(map);
```

## Screenshot
Leaflet locate link text

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -213,7 +213,7 @@ var map = new L.Map('map', {
 
 ## Developers
 
-Run the demo locally with `grunt serve` and then open [localhost:9000/demo/index.html](http://localhost:9000/demo/index.html).
+Run the demo locally with `grunt connect` and then open [localhost:9000/demo/index.html](http://localhost:9000/demo/index.html).
```
Comment (domoritz) [inline README.md]:
```suggestion
Run the demo locally with `yarn start` or `npm run start` and then open [localhost:9000/demo/index.html](http://localhost:9000/demo/index.html).
```

File: src/L.Control.Locate.js [hunk 2]
```diff
@@ -283,6 +285,14 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
                 link.role = 'button';
                 link.href = '#';
                 var icon = L.DomUtil.create(options.iconElementTag, options.icon, link);
+
+                if(options.strings.text !== undefined) {
```
Comment (domoritz) [inline src/L.Control.Locate.js]:
```suggestion
                if (options.strings.text !== undefined) {
```

File: src/L.Control.Locate.js [hunk 3]
```diff
@@ -283,6 +285,14 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
...
+                    var text = L.DomUtil.create(options.textElementTag, 'leaflet-locate-text', link);
+                    text.textContent = '\xa0' + options.strings.text;
```
Comment (domoritz) [inline src/L.Control.Locate.js]: Why do you need to add `\xa0`?

File: src/L.Control.Locate.js [hunk 4]
```diff
@@ -283,6 +285,14 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
...
+		    link.classList.add('leaflet-locate-text-active');
```
Comment (domoritz) [inline src/L.Control.Locate.js]: indentation?

File: src/L.Control.Locate.scss [hunk 5]
```diff
@@ -15,6 +15,15 @@
   }
 }
 
+.leaflet-touch .leaflet-bar .leaflet-locate-text-active  {
```
Comment (domoritz) [inline src/L.Control.Locate.scss]: What does this have to do with the proposed change?

**Review by domoritz — APPROVED** [review]
Thanks for the pull request. Can you add a screenshot to the description?

File: src/L.Control.Locate.js (hunk 3 above)
Comment (yanis-sadeg) [reply src/L.Control.Locate.js]: > 
> 
> Why do you need to add `\xa0`?

To make a space between the icon and the text, otherwise they are stuck together and it's ugly. I had the choice to do it by css or like this, i choosen this method. But thinking about, maybe I should make the space by css on the icon with a padding right, like this if the developer choose to use only the text without the icon there is no space for nothing. What do you think?

File: src/L.Control.Locate.js (hunk 4 above)
Comment (yanis-sadeg) [reply src/L.Control.Locate.js]: > 
> 
> indentation?

Yes... In my side it's correctly indented, I don't know why the indentation on this preview is wrong. I will check about it for the next push.

File: src/L.Control.Locate.scss (hunk 5 above)
Comment (yanis-sadeg) [reply src/L.Control.Locate.scss]: This allows to modify the css only if the text option is used, it will not impact the button if the option is not used. Also, I don't use the .leaflet-control-locate because otherwise I would have to use !important on the width property.

Modifications have been made on the css if the text option is displayed because of the default properties of leaflet which made it impossible to display the text correctly.

**Comment by yanis-sadeg** [comment]
> 
> 
> Thanks for the pull request. Can you add a screenshot to the description?

Sure, I will on the next push.

Edite: The requested changes have been made.

**Comment by domoritz** [comment]
I meant a screenshot in the pull request description not the code. I just want some documentation for the feature here. Sorry for being ambiguous.

**Comment by yanis-sadeg** [comment]
> 
> 
> I meant a screenshot in the pull request description not the code. I just want some documentation for the feature here. Sorry for being ambiguous.

My bad, I had misunderstood. Done 👌

File: src/L.Control.Locate.js (hunk 3 above)
Comment (domoritz) [reply src/L.Control.Locate.js]: Yep, let's do it in CSS.

File: src/L.Control.Locate.js (hunk 4 above)
Comment (domoritz) [reply src/L.Control.Locate.js]: Make sure you use spaces.

File: src/L.Control.Locate.js [hunk 6]
```diff
@@ -283,6 +285,17 @@ You can find the project at: https://github.com/domoritz/leaflet-locatecontrol
                 link.role = 'button';
                 link.href = '#';
                 var icon = L.DomUtil.create(options.iconElementTag, options.icon, link);
+
+                if (options.strings.text !== undefined) {
+                    var text = L.DomUtil.create(options.textElementTag, 'leaflet-locate-text', link);
+                    text.textContent = options.strings.text;
+		            link.classList.add('leaflet-locate-text-active');
+                    link.parentNode.style.display = "flex";
+                    if(options.icon.length > 0) {
```
Comment (domoritz) [inline src/L.Control.Locate.js]:
```suggestion
                    if (options.icon.length > 0) {
```

**Comment by domoritz** [comment]
Thank you for the addition!
